src/services/sdk/client-builder.ts

```diff
@@ -0,0 +1,54 @@
+import {
+  Client,
+  ClientBuilder,
+  type HttpMiddlewareOptions,
+} from '@commercetools/sdk-client-v2';
+import { getToken } from '../http/get-token-from-cookie';
+
+const AUTH_URL = import.meta.env['VITE_AUTH_URL'];
+const API_URL = import.meta.env['VITE_API_URL'];
+const CLIENT_SECRET = import.meta.env['VITE_CLIENT_SECRET'];
+const CLIENT_ID = import.meta.env['VITE_CLIENT_ID'];
+const SCOPES = import.meta.env['VITE_SCOPES'];
+const PROJECT_KEY = import.meta.env['VITE_PROJECT_KEY'];
+
+const projectKey = PROJECT_KEY;
+const scopes = SCOPES.split(' ');
+
+const httpMiddlewareOptions: HttpMiddlewareOptions = {
+  host: API_URL,
+  fetch,
+};
+
+export const ctpClient = (): Client => {
+  const [accessToken, refreshToken] = getToken();
```

Comment:
This solution looks very unreliable! Cookies can contain many different values. You should get accessToken and refreshToken values ​​from cookies strictly by the name under which you saved them there.